src/main/java/dk/dbc/updateservice/update/UpdateServiceCore.java

```diff
@@ -205,6 +216,104 @@ public UpdateRecordResponseDTO updateRecord(UpdateServiceRequestDTO updateServic
+
+            RecordDTOMapper.toRecord(recordEntryDTO, rawRecord);
+
+            // Set relations for active record. Deleted record don't have any relations
+            if (!rawRecord.isDeleted()) {
+                if (MarcXChangeMimeType.ENRICHMENT.equals(rawRecord.getMimeType())) {
+                    relationParents = Collections.singletonList(getCommonRecord(bibliographicRecordId, agencyId));
+                } else {
+                    // Look for referenced records
+                    final MarcRecord marcRecord = RecordDTOMapper.getMarcRecord(recordEntryDTO);
+                    relationParents = getParentRelations(marcRecord);
+                }
+            } else {
+                relationParents = Collections.emptyList();
+            }
+
+            final LibraryGroup libraryGroup = vipCoreService.getLibraryGroup(Integer.toString(agencyId));
+            final String providerId = EnqueueRecordAction.getProvider(settings, libraryGroup);
+            rawRepo.saveRecord(rawRecord);
+            rawRepo.removeLinks(recordId);
+            for (RecordId relation : relationParents) {
+                makeSureParentRecordExists(relation.getBibliographicRecordId(), relation.getAgencyId());
+                rawRepo.linkRecordAppend(recordId, relation);
+            }
+
+            rawRepo.enqueue(recordId, providerId, true, false, 1001);
+        } catch (JsonProcessingException | VipCoreException | RawRepoException ex) {
+            throw new UpdateException(ex.getMessage(), ex);
+        }
+    }
+
+    private void makeSureParentRecordExists(String bibliographicRecordId, int agencyId) throws UpdateException {
+        boolean shouldSave = false;
+        final Record parentRecord = rawRepo.fetchRecord(bibliographicRecordId, agencyId);
+        final Instant originalModified = parentRecord.getModified();
+        if (parentRecord.getMimeType() == null || parentRecord.getMimeType().isEmpty()) {
+            // In case it is a brand-new record
+            parentRecord.setMimeType(StoreRecordAction.getMarcXChangeMimetype(agencyId));
+            parentRecord.setContentJson("{}".getBytes());
+            shouldSave = true;
+        }
+        if (parentRecord.isDeleted()) {
+            // Record already exists but is deleted
+            parentRecord.setDeleted(false);
+            shouldSave = true;
+        }
+        if (shouldSave) {
+            parentRecord.setModified(originalModified);
+            rawRepo.saveRecord(parentRecord);
+        }
+    }
+
+    private RecordId getCommonRecord(String bibliographicRecordId, int agencyId) throws RawRepoException, VipCoreException, UpdateException {
+        if (agencyId == 191919) {
+            final List<Integer> agencyPriorityList = vipCoreService.getAgencyPriority(agencyId);
+            agencyPriorityList.remove(Integer.valueOf(agencyId)); // We have to use Integer, as the valueOf which takes an int uses the value as index not object
```

Comment:
I forrige system var agencyid en streng - det ville have sparet megen smerte hvis det stadigvk var det :(

Reply:
Det var generel jammer fra min side , ikke et udtryk for at du skulle lave det anderledes :-)